pkg/security/podsecuritypolicy/selinux/mustrunas.go

```diff
@@ -74,3 +81,128 @@ func (s *mustRunAs) Validate(fldPath *field.Path, _ *api.Pod, _ *api.Container,
 
 	return allErrs
 }
+
+func equalLevels(expectedLevel, actualLevel string) (bool, error) {
+	if expectedLevel == actualLevel {
+		return true, nil
+	}
+
+	// Level format: https://selinuxproject.org/page/MLSStatements#level
+
+	// "s0:c6,c0" => [ "s0", "c6,c0" ]
+	expectedParts := strings.SplitN(expectedLevel, ":", 2)
+	actualParts := strings.SplitN(actualLevel, ":", 2)
+
+	// "s0-s0" => [ "s0" ]
+	expectedSensitivity, err := canonicalizeSensitivity(expectedParts[0])
+	if err != nil {
+		return false, err
+	}
+
+	actualSensitivity, err := canonicalizeSensitivity(actualParts[0])
+	if err != nil {
+		return false, err
+	}
+
+	if !reflect.DeepEqual(expectedSensitivity, actualSensitivity) {
+		return false, nil
+	}
+
+	expectedPartsLen := len(expectedParts)
+	actualPartsLen := len(actualParts)
+
+	// both levels don't have categories and equal ("s0" and "s0")
+	if expectedPartsLen == 1 && actualPartsLen == 1 {
+		return true, nil
+	}
+
+	// "s0" != "s0:c1"
+	if expectedPartsLen != actualPartsLen {
+		return false, nil
+	}
+
+	// "c6,c0" => [ "c0", "c6" ]
+	expectedCategories, err := canonicalizeCategories(expectedParts[1])
+	if err != nil {
+		return false, err
+	}
+	actualCategories, err := canonicalizeCategories(actualParts[1])
+	if err != nil {
+		return false, err
+	}
+
+	// TODO: add support for dominance
+	// See: https://selinuxproject.org/page/NB_MLS#Managing_Security_Levels_via_Dominance_Rules
+	return reflect.DeepEqual(expectedCategories, actualCategories), nil
+}
+
+// Parses and canonicalize a sensitivity. Performs expansion (s0-s1 => s0,s1)
+// and simplification (s0-s0 => s0) if needed.
+func canonicalizeSensitivity(sensitivity string) ([]string, error) {
+	return canonicalizeItems(sensitivity, ",", "-", "s")
+}
+
+// Parses and canonicalize a categories. Performs expansion (c0-c1 => c0,c1) if needed.
+func canonicalizeCategories(categories string) ([]string, error) {
+	return canonicalizeItems(categories, ",", ".", "c")
+}
+
+func canonicalizeItems(str, itemsSeparator, rangeSeparator, boundaryPrefix string) ([]string, error) {
+	parts := strings.Split(str, itemsSeparator)
+	result := make([]string, 0, len(parts))
```

Comment:
nit: I personally wouldn't bother pre-allocating here, but your call.

Reply:
I'd prefer to leave it.